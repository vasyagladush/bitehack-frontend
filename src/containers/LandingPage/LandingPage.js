import React, { useState } from "react";
import classes from "./LandingPage.module.css";
import Navigation from "../../components/Navigation/Navigation";
import { Form, Button, Spinner } from "react-bootstrap";
import { Link } from "react-router-dom";
import axios from "axios";
import { url } from "../../ApiUrl.js";

const LandingPage = (props) => {
  const [state, setState] = useState({
    email: "",
    password: "",
    error: undefined,
    loading: false,
  });

  const inputHandler = (event) => {
    setState({
      ...state,
      [event.target.name]: event.target.value,
    });
  };

  const login = async () => {
    const response = await axios.post(url + "/auth/login", {
      email: state.email,
      password: state.password,
    });

    localStorage.setItem("token", response.data.token);
    window.location.href = "/";
  };

  return (
    <>
      <Navigation home history={props.history} />
      <div style={{ margin: "auto", marginTop: "30px", textAlign: "center" }}>
        <h1 style={{ fontSize: "2em" }}>Login</h1>
      </div>
      <div className={classes.LandingPage}>
        <Form style={{ marginTop: "25px" }}>
          <Form.Group controlId="formBasicEmail">
            <Form.Control
              type="text"
              placeholder="Enter Email"
              className={classes.input}
              style={{ width: "50%" }}
              name="email"
              onChange={inputHandler}
              value={state.email}
            />
          </Form.Group>
          <Form.Group style={{ marginTop: "25px" }} controlId="formBasicLogin">
            <Form.Control
              type="password"
              placeholder="Enter Password"
              className={classes.input}
              style={{ width: "50%" }}
              name="password"
              onChange={inputHandler}
              value={state.password}
            />
          </Form.Group>
          {state.loading ? (
            <Spinner animation="border" variant="dark" />
          ) : (
            <Button
              variant="dark"
              className={classes.button}
              style={{ marginTop: 20 + "px", marginBottom: 20 + "px" }}
              onClick={login}
            >
              Login
            </Button>
          )}
          {state.error ? <p>Wrong login or password</p> : null}
        </Form>
      </div>
      <div style={{ textAlign: "center", marginTop: 20 + "px" }}>
        <Link to="/register">Don't have an account?</Link>
      </div>
    </>
  );
};

export default LandingPage;
