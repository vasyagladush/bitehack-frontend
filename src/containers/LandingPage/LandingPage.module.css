.LandingPage {
    margin: auto;
    margin-top: 100px;
    text-align: center;
    border: 1px solid black;
    border-radius: 10px;
    width: 35%;
}

.image {
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 50px;
}

.input {
    margin: auto;
}

.button {
    margin-bottom: 10px;
}

@media (max-width: 650px){
    .LandingPage{
        width: 100%;
    }
}