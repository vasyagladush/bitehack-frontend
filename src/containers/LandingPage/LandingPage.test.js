import React from 'react'; 

import LandingPage from './LandingPage';
import {BrowserRouter as Router} from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import {configure} from 'enzyme';
import renderer from 'react-test-renderer';

configure({adapter: new Adapter});

describe('<LandingPage/>',()=>{
    it('matches snapshot',()=>{
        const wrapper = renderer.create(<Router>
            <LandingPage/>
        </Router>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});