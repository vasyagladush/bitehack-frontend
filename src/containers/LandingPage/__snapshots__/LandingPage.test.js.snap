// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage/> matches snapshot 1`] = `
Array [
  <div
    className="nav"
  >
    <p
      onClick={[Function]}
    >
      Social
    </p>
    <p
      className="active"
    >
      Home
    </p>
    <p
      onClick={[Function]}
    >
      About
    </p>
  </div>,
  <img
    className="image"
    height="300"
    src="messenger.png"
    width="300"
  />,
  <div
    className="LandingPage"
  >
    <form
      className=""
      style={
        Object {
          "marginTop": "25px",
        }
      }
    >
      <div
        className="form-group"
      >
        <input
          className="input form-control"
          id="formBasicEmail"
          name="login"
          onChange={[Function]}
          placeholder="Enter Login"
          style={
            Object {
              "width": "50%",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="input form-control"
          id="formBasicLogin"
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          style={
            Object {
              "width": "50%",
            }
          }
          type="text"
          value=""
        />
      </div>
      <button
        className="button btn btn-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </form>
  </div>,
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <a
      href="/register"
      onClick={[Function]}
    >
      You don't have account? CLICK!
    </a>
  </div>,
]
`;
